src/ingest_validation_tools/upload.py

```diff
@@ -592,32 +596,31 @@ def _constraint_checks(self, schema: SchemaVersion):
         except Exception:
             return self._get_constraint_check_errors(response, schema)
 
-    def _get_constrained_fields(self, schema: SchemaVersion):
+    def _get_constrained_fields(self, schema: SchemaVersion) -> list[str]:
```

Comment:
rather than building a dict where most things have the Entity API URL as a value, just collect field names at this point